### PR TITLE
Break out schemas into a single repo

### PR DESCRIPTION
**Pull request by lclarkmichalek**
This is so they can be referenced from qmail, nimbus, experience-ui, anywhere at all!

@buggyvelarde @havelkam

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+Schemas for qmail configs.
+
+## Usage
+
+```
+const schema = require('@qutics/qmail-schemas')
+const Ajv = require('ajv')
+
+const ajv = new Ajv()
+const valid = ajv.validate(schema, espConfig)
```
Comment (buggyvelarde) [inline README.md]: nice!

File: test/should-fail/no-spamEmailPattern.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (buggyvelarde) [inline test/should-fail/no-spamEmailPattern.yaml]: is this yaml?

File: test/should-fail/maximumInPeriod-period-too-low.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+expirySecs: 86400
```
Comment (buggyvelarde) [inline test/should-fail/maximumInPeriod-period-too-low.yaml]: this is yaml

File: test/should-fail/no-spamEmailPattern.yaml (hunk 2 above)
Comment (lclarkmichalek) [inline test/should-fail/no-spamEmailPattern.yaml]: JSON is valid YAML (with a few corner cases that aren't), so it works. These tests are all based around copying the default config and then removing lines, so I didn't rewrite them to be yaml

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+const spamSchema = require('./spam')
+
+module.exports = function configSchema () {
```
Comment (havelkam) [inline src/index.js]: I would prefer to specify the schemas in statical fashion as JSON files. I am especially not sure about exporting functions. While options are:

- JSON file
- export schema object
- export function

JSON file seems to me most reasonable to consume from other codebases. I might be able to figure out other pros and cons there.

Comment (lclarkmichalek) [inline src/index.js]: In the future we want to include schemas from integrations in here, and to do that we'd need to inspect the intergration implementations. That requires doing things dynamically, which is why I make this a function. I also map in src/spam/index.js, which wouldn't be possible with plain JSON. Plain JSON files would also require me to have the schema all in a single file, which isn't easy to work on.

In the future, I imagine we'll have config sections hidden behind feature flags, so the ability to parameterize the schema will be useful, and that requires exposing a function.

Comment (havelkam) [inline src/index.js]: I think that even if you will need import schema from integrations they could still all be in static files and you can just provide code to load these and handle them. There will be the benefit of being able to load just static scheme from somewhere else.

You are able to use definitions and refs to import schemas from separate files if you want to re-structure more complex ones: https://spacetelescope.github.io/understanding-json-schema/structuring.html

I am not sure about the idea of parametrising the schema, it seems to me that it would just make them more complex and more difficult to reason about.
